teps/0106-support-specifying-metadata-per-task-in-runtime.md

```diff
@@ -0,0 +1,261 @@
+---
+status: proposed
+title: Support Specifying Metadata per Task in Runtime
+creation-date: '2022-04-19'
+last-updated: '2022-04-19'
+authors:
+- '@austinzhao-go'
+---
+
+# TEP-0106: Support Specifying Metadata per Task in Runtime
+
+<!-- toc -->
+- [TEP-0106: Support Specifying Metadata per Task in Runtime](#tep-0106-support-specifying-metadata-per-task-in-runtime)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+    - [Vault Sidecar Injection](#vault-sidecar-injection)
+    - [Hermetically Executed Task](#hermetically-executed-task)
+    - [General Use Case](#general-use-case)
+  - [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+  - [Design Details](#design-details)
+    - [Metadata Precedence](#metadata-precedence)
+  - [Design Evaluation](#design-evaluation)
+    - [Reusability](#reusability)
+    - [Simplicity](#simplicity)
+    - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+    - [Add Metadata under `TaskRef` in `Pipeline`](#add-metadata-under-taskref-in-pipeline)
+    - [Create a `PipelineTaskRef` type](#create-a-pipelinetaskref-type)
+    - [Utilize Parameter Substitutions](#utilize-parameter-substitutions)
+  - [Test Plan](#test-plan)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+  - [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This work will support a user specifying the required metadata (annotations and/or labels) for a referenced `Task` in a `PipelineRun`. So the metadata depending on an execution context can be added in the runtime when they can not be statically defined in a `Task` during the "authoring" time.
+
+## Motivation
+
+The required metadata currently can be added under a `Task` entity while a user is authoring/configuring the template for a `TaskRun`. As two contexts are considered for Tetkon - the “authoring time” and “runtime”, a stretch of thinking will lead to if the metadata could be defined “dynamically” under the runtime.
```

Comment:
```suggestion
The required metadata currently can be added under a `Task` entity while a user is authoring/configuring the template for a `TaskRun`. As two contexts are considered for Tekton - the “authoring time” and “runtime”, a stretch of thinking will lead to if the metadata could be defined “dynamically” under the runtime.
```

Reply:
thanks @imjasonh for the check-in! 
parked in community slack for few days, seems no further feedback at this moment.

could I have have a lgtm label when good with u? will keep a tracking and update with the coming changes PR on this TEP.

cc @dibyom